clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -483,12 +484,21 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
                         parsed.add(parseRecord(tp, logEntry));
                         bytes += logEntry.size();
                     }
+
                     if (!parsed.isEmpty()) {
                         ConsumerRecord<K, V> record = parsed.get(parsed.size() - 1);
                         this.subscriptions.fetched(tp, record.offset() + 1);
                         this.records.add(new PartitionRecords<K, V>(fetchOffset, tp, parsed));
                         this.sensors.recordsFetchLag.record(partition.highWatermark - record.offset());
+                    } else if (buffer.capacity() >= this.fetchSize) {
+                        // we did not read a single message from a max fetchable buffer
+                        // because that message's size is larger than fetch size, in this case
+                        // throw the record too large exception
+                        throw new RecordTooLargeException("There is a single message whose size " +
```

Comment:
Since this is thrown from the fetch response handler, is it possible that it could be raised in one of KafkaConsumer's blocking methods such as commitSync?